### PR TITLE
AppUsers create

### PR DESCRIPTION
**Pull request by lemieux**
@alavers I wasn't really sure how to make this only available to node, but I figure it doesn't really matter as long as it only accepts the jwt authenticate.

### CODE REVIEW

File: src/api/base.js [hunk 1]
```diff
@@ -31,10 +31,11 @@ export class BaseApi {
 
   /**
    * Validates the headers sent to the server
+   * @param  {array} allowedAuth  - an array of allowedAuth to override the ones on the instance
    * @return {object}             - the headers object passed in parameter
    */
-  validateAuthHeaders() {
-    if (!this.allowedAuth || this.allowedAuth.length === 0) {
+  validateAuthHeaders(allowedAuth = this.allowedAuth) {
```
Comment (lemieux) [inline src/api/base.js]: this thing is soooo awesome

File: src/api/appUsers.js [hunk 2]
```diff
@@ -24,6 +24,30 @@ export class AppUsersApi extends BaseApi {
     });
   }
 
+  create(userId, props = {}) {
+    if (!userId || !userId.trim()) {
+      throw new Error('Must provide an user id.');
```
Comment (alavers) [inline src/api/appUsers.js]: You should name the parameter specifically in this error, eg `userId` not `user id`.

Comment (lemieux) [inline src/api/appUsers.js]: fixed.

File: src/api/appUsers.js [hunk 3]
```diff
@@ -24,6 +24,30 @@ export class AppUsersApi extends BaseApi {
...
+    }
+
+    let payload = Object.assign({
+      userId: userId
+    }, props);
+
+    if (props.signedUpAt instanceof Date) {
```
Comment (alavers) [inline src/api/appUsers.js]: JSON serialization will do this for you. In fact, you should reject a signedUpAt param if it isn't a `Date`

**Comment by alavers** [comment]
Rest looks :+1:

File: src/api/appUsers.js (hunk 3 above)
Comment (lemieux) [inline src/api/appUsers.js]: Ok perfect, I'll throw if it's not a date

**Pull request merged**